flask_monitoringdashboard/core/exception_logger.py

```diff
@@ -0,0 +1,108 @@
+import inspect
+import traceback
+import hashlib
+
+from types import FrameType, TracebackType
+from typing import Union
+
+from sqlalchemy.orm import Session
+from flask_monitoringdashboard.core.user_exception_logger import ExcInfo, ScopedExceptionLogger
+from flask_monitoringdashboard.database import CodeLine, FunctionDefinition
+from flask_monitoringdashboard.database.exception_info import add_exception_info
+from flask_monitoringdashboard.database.stack_trace_snapshot import add_stack_trace_snapshot, get_stack_trace_by_hash
+from flask_monitoringdashboard.database.exception_stack_line import add_exception_stack_line
+from flask_monitoringdashboard.database.function_definition import add_function_definition
+from flask_monitoringdashboard.database.exception_message import add_exception_message
+from flask_monitoringdashboard.database.exception_type import add_exception_type
+
+
+def hash_helper(s: str):
+    return hashlib.sha256(s.encode('utf-8')).hexdigest()
+
+def h_chain(h: str, tb: Union[TracebackType, None]):
+    if tb is None:
+        return h
+
+    f_def = get_function_definition_from_frame(tb.tb_frame)
+    new_hash = hash_helper(h+f_def.function_hash)
+
+    return h_chain(new_hash, tb.tb_next)
+
+def get_function_definition_from_frame(frame: FrameType) -> FunctionDefinition:
+    f_def = FunctionDefinition()
+    f_def.function_code = inspect.getsource(frame.f_code)
+    f_def.function_hash = hash_helper(f_def.function_code)
+    return f_def
+
+def create_codeline_from_frame(frame: FrameType, lineno):
+    c_line = CodeLine()
+    c_line.filename = frame.f_code.co_filename
+    c_line.line_number = lineno
+    c_line.function_name = frame.f_code.co_name
+    code_context = inspect.getframeinfo(frame).code_context
+    if code_context is not None and len(code_context) > 0:
+        c_line.code = code_context[0]
+    return c_line
+
+def hash_stack_trace(exc, tb):
```

Comment:
add small comment here? If i understand correctly the hashed stack trace is the hash of the exception name (?) concatenated to the hash of the traceback object, line by line?